src/clarity-angular/data/tree-view/tree-node.ts

```diff
@@ -1,142 +1,189 @@
+    ]
+
 })
-export class TreeNode extends AbstractTreeSelection {
+export class TreeNode
+    extends AbstractTreeSelection
+    implements OnDestroy {
+
+    constructor(public nodeExpand: Expand,
+                @Optional() @SkipSelf() public parent: TreeNode,
+                public treeSelectionService: TreeSelectionService) {
+        super(parent);
+        if (this.parent) {
+            this.parent.register(this);
+        }
+    }
 
-    constructor(@Optional() @SkipSelf() parent: TreeNode,
-                @Optional() @SkipSelf() public treeSelectionService: TreeSelectionService) {
-        super(parent, treeSelectionService);
+    private _children: TreeNode[] = [];
+
+    get children(): TreeNode[] {
+        return this._children;
     }
 
-    @Input("clrTreeModel") model: any;
+    /* Registration */
 
-    @ContentChildren(TreeNode) _children: QueryList<TreeNode>;
+    checkIfChildNodeRegistered(node: TreeNode): boolean {
+        return (this.children.indexOf(node) > -1);
+    }
 
-    /**
-     * Generates the child TreeNodes array from the ContentChildren QueryList
-     * @returns {TreeNode[]|Array}
-     */
-    get children(): AbstractTreeSelection[] {
-        return this._children ? this._children.toArray().filter(child => this !== child) : [];
+    //TODO: This should ideally be in AbstractTreeSelection
+    //Tried doing this but ran into some issues and also ran out of time.
+    //Will get this done later.
+    register(node: TreeNode): void {
+        if (!this.checkIfChildNodeRegistered(node)) {
+            this.children.push(node);
+            if (this.selectable) {
+                if (this.selected) {
+                    node.parentChanged(this.selected);
+                }
+            }
+        }
     }
 
-    get nodeSelected(): boolean {
-        return this.selected;
+    //TODO: This should ideally be in AbstractTreeSelection
+    //Tried doing this but ran into some issues and also ran out of time.
+    //Will get this done later.
+    unregister(node: TreeNode): void {
+        const index = this.children.indexOf(node);
+        if (index > -1) {
+            this.children.splice(index, 1);
+        }
     }
 
-    @Input("clrTreeNodeSelected")
-    set nodeSelected(value: boolean) {
+    /* Selection */
+
+    private _selectionActivated: boolean = false;
+
+    activateSelection(): void {
+        if (!this._selectionActivated && this.treeSelectionService && !this.treeSelectionService.selectable) {
+            this.treeSelectionService.selectable = true;
+            this._selectionActivated = true;
+        }
+    }
+
+    @Input("clrSelected")
+    public set nodeSelected(value: boolean) {
+        //required for recursive trees to discard unset inputs.
+        if (value === undefined || value === null) {
+            return;
+        }
         if (this.selected !== value) {
             this.selected = value;
         }
+        this.activateSelection();
```

Comment:
I just realized you are not activating selection if the only bindings are set to `undefined` initially. You should move the activation at the very top of the method.